tensorflow/core/lib/strings/numbers.h

```diff
@@ -122,6 +122,12 @@ bool safe_strtof(const char* str, float* value);
 // Values may be rounded on over- and underflow.
 bool safe_strtod(const char* str, double* value);
 
+// Convert strings to number of type T.
+// Leading and trailing spaces are allowed.
+// Values may be rounded on over- and underflow.
+template <typename T>
+bool SafeStringToNumeric(StringPiece s, T* value);
```

Comment:
I'm not 100% sure, but I think this might cause linker problems with some compilers:

https://isocpp.org/wiki/faq/templates#templates-defn-vs-decl

Perhaps `extern template` is necessary?

https://stackoverflow.com/a/8131212/3574081